zipline/algorithm.py

```diff
@@ -383,6 +384,40 @@ def __init__(self, *args, **kwargs):
 
         self.benchmark_sid = kwargs.pop('benchmark_sid', None)
 
+        self.capital_changes = kwargs.pop('capital_changes', {})
+        if self.capital_changes:
+            self.capital_changes = \
```

Comment:
> It could be that zipline works directly with the appropriate date object, and we take care of parsing strings to dates upstream

Strong :+1: on moving this logic upstream (and removing all the non-midnight date normalization if possible).

As a general note, the question of "what serialization format should we use to do IPC at Quantopian?" is very different from "what interface should TradingAlgorithm present for providing this data".  We should make sure that we're clear about what layer of abstraction we're interested in when we're talking about how we represent data.